### PR TITLE
Attendance Command

### PR DESCRIPTION
**Pull request by baconlover2020**
Hey guys
I had to add another intent (voice_states) for this to work.
Also, I added it to the cog StudentCommands because it seems to me it is the most reasonable option.
I can add it somewhere else or create a separate cog if requested.

### CODE REVIEW

File: Cogs/StudentCommands.py [hunk 1]
```diff
@@ -144,3 +144,20 @@ async def ping(self, ctx):
         latency = round(self.bot.latency * 1000)
         await ctx.send(f'{latency} ms')
         await log(self.bot, f'{ctx.author} pinged from #{ctx.channel}, response took {latency} ms')
+
+    @commands.command()
+    async def attendance(self, ctx):
+
+        # The user object does not have a .voice.channel attribute so simply using author.id.voice does not work here
+        try:
+            channel = ctx.guild.get_member(ctx.message.author.id).voice.channel
```
Comment (CodingPenguin1) [inline Cogs/StudentCommands.py:153]: If you ever need to get a specific member object, please_ use get_member() from utils.py. The builtin get_member() function is a direct API call and can take a long time to run.

Also note that the context object provides
`discord.ext.commands.Context.author.voice.channel`
This should negate the need for a get_member() call entirely

File: Cogs/StudentCommands.py [hunk 2]
```diff
@@ -144,3 +144,20 @@ async def ping(self, ctx):
...
+            members = channel.members
+            members.remove(ctx.author)
+            if members:
+                attendees = "\n".join([member.name for member in members])
```
Comment (CodingPenguin1) [inline Cogs/StudentCommands.py:157]: Make the @ mentions please so that we can track a user across nickname changes

**Comment by CodingPenguin1** [comment]
@baconlover2020 

StudentCommands is a good place for this.

In the future, please document more in the PR. Ex. a title more specific than "Development" and a description that states the changes.

Also note that you can use this pane:
image
to request reviewers (I'm a good primary reviewer for this repo, and @tux2603 is a good second choice),
set assignees (set yourself),
set labels (enhancement for a new feature like this),
assign the PR to projects (unused feature in this repo),
assign the PR to milestones (unused feature in this repo),
and link issues (link the issues the PR resolves so that the issues automatically close when the PR is merged)

**Comment by baconlover2020** [comment]
@CodingPenguin1 
Alrighty! I will keep discord.utils in mind. Also, your suggestion that negates its use entirely worked great.
Thank you for the heads up about Github usage, I am still quite new to this! I really appreciate it.

File: Cogs/StudentCommands.py [hunk 3]
```diff
@@ -144,3 +144,20 @@ async def ping(self, ctx):
...
+        try:
+            channel = ctx.message.author.voice.channel
+            members = channel.members
+            members.remove(ctx.author)
+            if members:
+                attendees = "\n".join([member.mention for member in members])
+                await ctx.message.channel.send(f"Attendees of {channel.name} required by {ctx.message.author.mention}:\n {attendees}")
+            else:
+                await ctx.message.channel.send("There are no users in your voice channel.")
+        except:
```
Comment (CodingPenguin1) [inline Cogs/StudentCommands.py:161]: Cam you make this except handle the particular exception that gets raised if the requester isn't in a vc? There's an on_command_error handle on bot.py that would get clobbered by this all-catching except

**Review by CodingPenguin1 — CHANGES_REQUESTED** [review]

File: Cogs/StudentCommands.py (hunk 3 above)
Comment (baconlover2020) [reply Cogs/StudentCommands.py:161]: Sure! Will do.

**Review by CodingPenguin1 — APPROVED** [review]

**Pull request merged**